### PR TITLE
STAC-15121 Fix pipeline

### PR DESCRIPTION
**Pull request by julianosk**
### Step 3: Did you add / update tests for your changes in the right area?
- [ ] Unit/Component test
- [ ] Integration test
- [ ] E2E/Molecule test

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Did you add release notes describing the changes you made?
- [ ] Yes

### Step 6: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: .gitlab-ci-agent.yml [hunk 1]
```diff
@@ -235,28 +235,6 @@ unit_tests:
...
-    # TODO: --python-runtimes "$PYTHON_RUNTIMES"
-    - inv -e rtloader.test
-    - inv -e test --coverage --race --profile --fail-on-fmt --cpus 4 --major-version $MAJOR_VERSION --python-runtimes $PYTHON_RUNTIMES --build-include docker,linux,kubelet,kubeapiserver,python,orchestrator,clusterchecks,zlib
-  tags:
-    - sts-aws
-  rules:
-    - *cluster_agent
-  retry:
-    max: 2
-
```
Comment (LouisParkin) [inline .gitlab-ci-agent.yml:259]: Why is this deleted?  Are the cluster agent unit tests no longer relevant? Or have they perhaps been moved?

File: .assistance/Vagrantfile [hunk 2]
```diff
@@ -45,9 +45,9 @@ Vagrant.configure("2") do |config|
     echo "export GOLANG_VERSION=1.14" >> /home/vagrant/.env
     echo "export RVM_VERSION=1.29.12" >> /home/vagrant/.env
     echo "export RUBY_VERSION=2.4.2" >> /home/vagrant/.env
-    echo "export ARTIFACTORY_USER=user" >> /home/vagrant/.env
-    echo "export ARTIFACTORY_PASSWORD=ZPxcNe6dXZmGmCz3" >> /home/vagrant/.env
-    echo "export ARTIFACTORY_URL=artifactory.stackstate.io/artifactory/api/pypi/pypi-local" >> /home/vagrant/.env
+    echo "export artifactory_user=$artifactory_user" >> /home/vagrant/.env
+    echo "export artifactory_password=$artifactory_password" >> /home/vagrant/.env
```
Comment (LouisParkin) [inline .assistance/Vagrantfile:49]: Wouldn't this just assign existing ENV vars back to themselves?

File: .gitlab-scripts/setup_artifactory.cmd [hunk 3]
```diff
@@ -1,4 +1,4 @@
 if exist %APPDATA%\pip rd /s/q %APPDATA%\pip
 mkdir %APPDATA%\pip
 echo [global] > %APPDATA%\pip\pip.ini
-echo extra-index-url = https://%ARTIFACTORY_USER%:%ARTIFACTORY_PASSWORD%@%ARTIFACTORY_URL%/simple >> %APPDATA%\pip\pip.ini
+echo extra-index-url = https://%artifactory_user%:%artifactory_password%@%ARTIFACTORY_PYPI_URL% >> %APPDATA%\pip\pip.ini
```
Comment (LouisParkin) [inline .gitlab-scripts/setup_artifactory.cmd:4]: Does the pip.ini file live on public infrastructure? If so, does it get destroyed when it is not needed anymore?

File: .gitlab-scripts/setup_artifactory.sh [hunk 4]
```diff
@@ -1,4 +1,4 @@
 ## python artifactory dependency
 mkdir ~/.pip/ && touch ~/.pip/pip.conf
 echo "[global]" > ~/.pip/pip.conf
-echo "extra-index-url = https://$ARTIFACTORY_USER:$ARTIFACTORY_PASSWORD@$ARTIFACTORY_URL/simple" >> ~/.pip/pip.conf
+echo "extra-index-url = https://$artifactory_user:$artifactory_password@$ARTIFACTORY_PYPI_URL" >> ~/.pip/pip.conf
```
Comment (LouisParkin) [inline .gitlab-scripts/setup_artifactory.sh:4]: Same question as for setup_artifactory.cmd file

File: Development.md [hunk 5]
```diff
@@ -18,9 +18,9 @@ Remember to `git pull` every time you push a change.
 
 We use some private python libraries for our integrations therefore you need to configure artifactory as pypi repository:
 ```bash
-$ export ARTIFACTORY_URL=artifactory.stackstate.io/artifactory/api/pypi/pypi-local && \
-  export ARTIFACTORY_USER=... && \
-  export ARTIFACTORY_PASSWORD=... && \
+$ export ARTIFACTORY_PYPI_URL=artifactory.tooling.stackstate.io/artifactory/api/pypi/pypi-local/simple && \
+  export artifactory_user=... && \
+  export artifactory_password=... && \
```
Comment (LouisParkin) [inline Development.md:23]: Why make the variable names lowercase?

File: local_build.sh [hunk 6]
```diff
@@ -26,9 +26,9 @@ if [[ "$#" -eq "1" ]]; then
         MOUNT="/stackstate-agent-mount"
 
         docker run --rm \
-            -e ARTIFACTORY_USER=$ARTIFACTORY_USER \
-            -e ARTIFACTORY_PASSWORD=$ARTIFACTORY_PASSWORD \
-            -e ARTIFACTORY_URL="artifactory.stackstate.io/artifactory/api/pypi/pypi-local" \
+            -e artifactory_user=$artifactory_user \
+            -e artifactory_password=$artifactory_password \
```
Comment (LouisParkin) [inline local_build.sh:30]: Would this not just assign the value of the ENV vars back to themselves?

File: test/molecule-role/molecule/_shared/helm/install-stackstate-agent.yml [hunk 7]
```diff
@@ -7,11 +7,11 @@
...
     --set-string 'stackstate.url'='http://{{ hostvars['kubernetes-cluster-agent']['ansible_host'] }}:7077/stsAgent' \
     --set "agent.image.tag={{agent_current_branch}}" \
-    --set "agent.image.repository=docker.io/stackstate/{{agent_docker_repo}}" \
+    --set "agent.image.repository=stackstate/{{agent_docker_repo}}" \
     --set "clusterAgent.image.tag={{agent_current_branch}}" \
-    --set "clusterAgent.image.repository=docker.io/stackstate/{{cluster_agent_docker_repo}}" \
+    --set "clusterAgent.image.repository=stackstate/{{cluster_agent_docker_repo}}" \
     --set "clusterChecks.image.tag={{agent_current_branch}}" \
-    --set "clusterChecks.image.repository=docker.io/stackstate/{{agent_docker_repo}}" \
+    --set "clusterChecks.image.repository=stackstate/{{agent_docker_repo}}" \
```
Comment (LouisParkin) [inline test/molecule-role/molecule/_shared/helm/install-stackstate-agent.yml:14]: Why does the repository variable no longer include docker.io?  Is it to be able to add it later on (perhaps substitute with quay.io)?

**Review by LouisParkin — COMMENTED** [review]
Hi @julianosk 
In principle, I am happy to approve this PR if it works, I do have some questions though.

File: pkg/util/kubernetes/apiserver/common/common.go [hunk 8]
```diff
@@ -3,6 +3,7 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-2020 Datadog, Inc.
 
+//go:build kubeapiserver
```
Comment (TizianoPerrucci) [inline pkg/util/kubernetes/apiserver/common/common.go:6]: this is not needed

File: .assistance/Vagrantfile (hunk 2 above)
Comment (TizianoPerrucci) [reply .assistance/Vagrantfile:49]: this is not setting back to itself, it is setting it in the vagrant env file
the vagrant vm does not inherit your local env variables

File: .gitlab-ci-agent.yml (hunk 1 above)
Comment (TizianoPerrucci) [reply .gitlab-ci-agent.yml:259]: we figured it was redundant, runs the same test as the unit_test step

File: .gitlab-scripts/setup_artifactory.cmd (hunk 3 above)
Comment (TizianoPerrucci) [reply .gitlab-scripts/setup_artifactory.cmd:4]: this runs it in the gitlab runner, check the gitlab-ci-agent

File: local_build.sh (hunk 6 above)
Comment (TizianoPerrucci) [reply local_build.sh:30]: the docker container does not inherit the variables from your local, you need to explicitly set it

File: test/molecule-role/molecule/_shared/helm/install-stackstate-agent.yml (hunk 7 above)
Comment (TizianoPerrucci) [reply test/molecule-role/molecule/_shared/helm/install-stackstate-agent.yml:14]: I found that out lately as well, I think the helm charts now include pulling from docker.io

**Review by TizianoPerrucci — APPROVED** [review]

File: Development.md (hunk 5 above)
Comment (Q69K) [reply Development.md:23]: this is from StackVista group CI variables

File: pkg/util/kubernetes/apiserver/common/common.go (hunk 8 above)
Comment (Q69K) [reply pkg/util/kubernetes/apiserver/common/common.go:6]:
```suggestion
```

**Review by Q69K — APPROVED** [review]

**Review by LouisParkin — APPROVED** [review]
